lib/Xmds/Soap7.php

```diff
@@ -291,7 +292,21 @@ public function GetWeather($serverKey, $hardwareKey): string
             throw new \SoapFault('Receiver', 'This Display is not authorised.');
         }
 
-        // TODO actually get the Weather with Display lat/long
-        return '{}';
+        $latitude = $this->display->latitude;
+        $longitude = $this->display->longitude;
+
+        // check for coordinates if present
+        if ($latitude && $longitude) {
+            // Dispatch an event to initialize weather data.
+            $event = new XmdsWeatherRequestEvent($latitude, $longitude);
+            $this->getDispatcher()->dispatch($event, XmdsWeatherRequestEvent::$NAME);
+        } else {
+            return json_encode([
```

Comment:
This is good, but it would be more consistent for SOAP to throw a SoapFault containing this message